File: quiz_app_flutter/lib/presentation/dashboard/intent/dashboard_intent.dart
import 'package:quiz_app_flutter/domain/entities/dashboard/dashboard_data.dart';

abstract class DashboardIntent {}

class LoadDashboard extends DashboardIntent {}

class NavigateToQuizSets extends DashboardIntent {
  final DashboardDataItem
  item; // Replace with your DashboardData.Item equivalent
  NavigateToQuizSets(this.item);
}

File: quiz_app_kmp/shared/src/commonMain/kotlin/com/tnm/quizappkmp/shared/presentation/dashboard/intent/DashboardIntent.kt
package com.tnm.shared.presentation.dashboard.intent

import com.tnm.quizappkmp.domain.DashboardData

sealed class DashboardIntent {
    object LoadDashboard : DashboardIntent()
    data class NavigateToQuizSets(val item: DashboardData.Item) : DashboardIntent()
}

sealed class DashboardNavEvent {
    data class NavigateToQuizSets(val item: DashboardData.Item) : DashboardNavEvent()
}